internal/kuberesource/mutators.go

```diff
@@ -201,38 +258,18 @@ func PatchNamespaces(resources []any, namespace string) []any {
 // PatchServiceMeshAdminInterface activates the admin interface on the
 // specified port for all Service Mesh components in a set of resources.
 func PatchServiceMeshAdminInterface(resources []any, port int32) []any {
+	var out []any
 	for _, resource := range resources {
-		switch r := resource.(type) {
-		case *applyappsv1.DeploymentApplyConfiguration:
-			for i := 0; i < len(r.Spec.Template.Spec.InitContainers); i++ {
-				// TODO(davidweisse): find service mesh containers by unique name as specified in RFC 005.
-				if strings.Contains(*r.Spec.Template.Spec.InitContainers[i].Image, "service-mesh-proxy") {
-					r.Spec.Template.Spec.InitContainers[i] = *r.Spec.Template.Spec.InitContainers[i].
-						WithEnv(NewEnvVar("EDG_ADMIN_PORT", fmt.Sprint(port))).
-						WithPorts(
-							ContainerPort().
-								WithName("admin-interface").
-								WithContainerPort(port),
-						)
-					ingressProxyConfig := false
-					for j, env := range r.Spec.Template.Spec.InitContainers[i].Env {
-						if *env.Name == "EDG_INGRESS_PROXY_CONFIG" {
-							ingressProxyConfig = true
-							env.WithValue(fmt.Sprintf("%s##admin#%d#true", *env.Value, port))
-							r.Spec.Template.Spec.InitContainers[i].Env[j] = env
-							break
-						}
-					}
-					if !ingressProxyConfig {
-						r.Spec.Template.Spec.InitContainers[i].WithEnv(
-							NewEnvVar("EDG_INGRESS_PROXY_CONFIG", fmt.Sprintf("admin#%d#true", port)),
-						)
-					}
-				}
+		out = append(out, MapPodSpecWithMeta(resource, func(meta *applymetav1.ObjectMetaApplyConfiguration, spec *applycorev1.PodSpecApplyConfiguration) *applycorev1.PodSpecApplyConfiguration {
+			_, ingressOk := meta.Annotations[smIngressConfigAnnotationKey]
+			_, egressOk := meta.Annotations[smEgressConfigAnnotationKey]
+			if ingressOk || egressOk {
+				meta.WithAnnotations(map[string]string{smAdminInterfaceAnnotationKey: fmt.Sprint(port)})
```

Comment:
I'm not too happy that this relies on the side effect that the input is mutable. This is also true for most other functions in this package, which claim to return an object but actually do updates in-place, too. But here it's a bit too much for my taste, given that `meta` is not even an output.

Reply:
We could update the `MapPodSpecWithMeta` function to also update the ObjectMeta and not only taking it as an input.